string behaviour
---
int16 index